### PR TITLE
Add offline database :airplane:

### PR DESCRIPTION
**Pull request by kosssi**
This feature allow to create pouch database on client side to facilitate offline first.

- [x] Create databases on init for each docType you want and pouchdb options
- [x] Create database after init for one docType and pouchdb options
- [x] Add unit test
- [x] Build library
- [ ] Update documentation

### CODE REVIEW

File: src/offline.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const PouchDB = require('pouchdb')
+
+export function init (cozy, offline) {
```
Comment (enguerran) [inline src/offline.js]: We could save 2 lines with ES2015 syntax, couldn´t we ?

```js
export function init (cozy, { options = {}, docTypes = [] }) {
  for (let docType of docTypes) {
    addDocType(cozy, docType, options)
  }
}
```

File: src/offline.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  const options = offline.options || {}
+  const docTypes = offline.docTypes || []
+  for (let docType of docTypes) {
+    addDocType(cozy, docType, options)
+  }
+}
+
+export function addDocType (cozy, docType, options) {
+  if (cozy._offline_databases === false) {
```
Comment (enguerran) [inline src/offline.js]: We may check if it is an array, is there another way to check that like `if (!Array.isArray(cozy._offline_databases) { cozy._offline_databases = [] }`

Comment (nono) [inline src/offline.js]: Or `cozy._offline_databases = cozy._offline_databases || []`

**Comment by aenario** [comment]
Looks like there is some conflict in leveldown installation between cozy-data-system & pouchdb ... 

You can try : 
- using yarn to install both
- installing data-system & proxy with --global

Good luck :clinking_glasses:

File: package.json [hunk 3]
```diff
@@ -61,5 +64,8 @@
     "All": {
       "description": "Test with any doctype."
     }
+  },
+  "dependencies": {
+    "pouchdb": "6.1.1"
```
Comment (enguerran) [inline package.json]: @kosssi could you explain why you had this dependency? I though we let the third-party application to add or not pouchdb in its dependencies.

File: webpack.config.js [hunk 4]
```diff
@@ -45,6 +45,9 @@ var config = {
   resolve: {
     root: path.resolve('./src'),
     extensions: ['', '.js']
+  },
+  externals: {
+    pouchdb: 'PouchDB'
```
Comment (enguerran) [inline webpack.config.js]: @kosssi it may be a typo here, cf. https://github.com/cozy/cozy-client-js/pull/43

File: package.json (hunk 3 above)
Comment (enguerran) [inline package.json]: Maybe it has to be in `devDependencies` if tests used it. Or in `peerDependencies` if not. I am not a pro with that.

**Pull request merged**